File: PropertyManagementSystem/PropertyManagementSystem/Models/MaintenanceSchedule.cs
using System;

namespace PropertyManagementSystem.Models
{
    public class MaintenanceSchedule: BaseModel
    {        
        public DateTime PeriodStartDate { get; set; }
        public DateTime PeriodEndDate { get; set; }
        public float MaintenanceAmount { get; set; }
        public long MaintenanceAgreementId { get; set; }
        public MaintenanceAgreement MaintenanceAgreement { get; set; }
    }
}

File: PropertyManagementUI/src/app/Models/MaintenanceSchedule.ts
import { BaseModel } from "./BaseModel";
import { MaintenanceAgreement } from "./MaintenanceAgreement";

export interface MaintenanceSchedule extends BaseModel{
  periodStartDate:Date,
  periodEndDate: Date,
  maintenanceAmount: number,
  status: boolean,
  transDate: Date,
  maintenanceAgreementId: number,
  maintenanceAgreement: MaintenanceAgreement
}
